### PR TITLE
Set Up Interrogate to Track Docstrings Usage

### PR DESCRIPTION
**Pull request by ghickman**
This adds [interrogate](https://interrogate.readthedocs.io/en/latest/) to our dev tooling stack to track how many objects have docstrings.

[sc-159]

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,12 @@ exclude_lines = [
 
 [tool.coverage.html]
 
+[tool.interrogate]
+fail-under = 26
```
Comment (CarolineMorton) [inline pyproject.toml:32]: What does this mean?

**Review by CarolineMorton — APPROVED** [review]
Looks good to me

File: pyproject.toml (hunk 1 above)
Comment (ghickman) [reply pyproject.toml:32]: Interrogate will return a failure code on the command line if coverage is below 26% (mirroring how test coverage works).  We're currently just over that, so this is base level to start with!

**Comment by sebbacon** [comment]
Given we use the google python style guide's docstring conventions, is there some linter / checker option for that?

**Comment by ghickman** [comment]
This tool only checks that objects have docstrings, not which style we're using. There are other tools for that, @StevenMaude has collated a [selection of them](https://app.shortcut.com/ebm-datalab/story/162/use-a-standard-for-docstrings).

**Pull request merged**